.github/workflows/ci_tests.yml

```diff
@@ -128,7 +128,7 @@ jobs:
       run: tox -e ${{ matrix.tox_env }} -- ${{ matrix.toxposargs }}
     - name: Upload coverage to codecov
       if: ${{ contains(matrix.tox_env, '-cov') }}
-      uses: codecov/codecov-action@v3
+      uses: codecov/codecov-action@13ce06bfc6bbe3ecf90edbbf1bc32fe5978ca1d3  # v5.3.1
```

Comment:
I don't remember if jumping from v3 to v5 have breaking changes in terms of workflow inputs. If coverage checks start to fail, you should revisit.